gapic/samplegen/samplegen.py

```diff
@@ -86,48 +92,147 @@ class InvalidRequestSetup(SampleError):
     pass
 
 
-class DuplicateInputParameter(SampleError):
-    pass
+class Validator:
+    """
+    Class that validates samples.
 
+    Contains methods that validate different segments of a sample and maintains
+    state that's relevant to validation across different segments.
+    """
 
-class ResponseValidator:
     COLL_KWORD = "collection"
     VAR_KWORD = "variable"
     MAP_KWORD = "map"
     KEY_KWORD = "key"
     VAL_KWORD = "value"
     BODY_KWORD = "body"
 
-    def __init__(self, response):
+    def __init__(self):
+        """
+        Args:
+           response: list[dict{str:?}]: The structured data representing
+                                        the sample's response.
+        """
+
+        # The response ($resp) variable is special and guaranteed to exist.
+        self.var_defs_: Set[str] = {"$resp"}
+
+    # TODO: this will eventually need the method name and the proto file
+    # so that it can do the correct value transformation for enums.
+    def validate_and_transform_request(self,
+                                       request: List[Mapping[str, str]]) -> List[TransformedRequest]:
+        """Validates the "request" block from a sample config and returns a transformed
```

Comment:
nit: First doc string line should fit in a physical line (cf https://github.com/google/styleguide/blob/gh-pages/pyguide.md#381-docstrings)